libraries/provider_component_runit_supervisor_sysvinit.rb

```diff
@@ -24,9 +22,43 @@ def action_create
           end
         end
 
+        def action_delete
+          Dir["#{new_resource.install_path}/service/*"].each do |svc|
+            execute "#{new_resource.install_path}/embedded/bin/sv stop #{svc}" do
+              retries 5
+            end
+          end
+
+          ruby_block "remove inittab entry" do
+            block do
+              f = Chef::Util::FileEdit.new "/etc/inittab"
+              f.search_file_delete svdir_line
+              f.write_file
+            end
+            only_if "grep '#{svdir_line}' /etc/inittab"
+            notifies :run, "execute[init q]", :immediately
+            notifies :run, "execute[pkill -HUP -P 1 runsv$]", :immediately
+          end
+
+          execute "init q" do
+            action :nothing
+          end
+
+          # To avoid stomping on runsv's owned by a different runsvdir
+          # process, kill any runsv process that has been orphaned, and is
+          # now owned by init (process 1).
+          execute "pkill -HUP -P 1 runsv$" do
```

Comment:
This could still technically kill process we don't intend to.  Unfortunately it is the same process we currently use in the post-stop script for old upstart versions.